apps/website/src/pages/admin/bingo/[bingoId]/live.tsx

```diff
@@ -35,9 +36,18 @@ import { BingoCardGrid } from "@/components/bingo/BingoCardGrid";
 import { transposeMatrix } from "@/utils/math";
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const session = await getSession(context);
   const adminProps = await getAdminSSP(context, permissions.manageBingos);
   if (!adminProps) {
-    return { notFound: true };
+    const id = context.params?.bingoId;
+    return {
+      redirect: {
+        destination: session?.user?.id
+          ? "/unauthorized"
+          : `/auth/signin?callbackUrl=/admin/bingo${id ? "/" + id + "/live" : ""}`,
```

Comment:
I think it would be cleaner and more maintainable to use the context object to get the path (same for all routes probably). We could also try to refactor further so we do not have the same code in each route, but im happy with being explicit here.

```suggestion
          : `/auth/signin?callbackUrl=${encodeURIComponent(context.resolvedUrl)}`,
```